### PR TITLE
Configurable Codecs

### PR DESCRIPTION
**Pull request by redmerloen**
### Description of fix

Make the codecs configurable from the setup of the endpoint

### CODE REVIEW

File: Example/VialerSIPLib/AppDelegate.swift [hunk 1]
```diff
@@ -52,10 +52,15 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             return [VSLTransportConfiguration(transportType: .UDP)!]
         }
 
+
```
Comment (bobvoorneveld) [inline Example/VialerSIPLib/AppDelegate.swift]: enter to much

File: Pod/Classes/VSLCodecs.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    // Speex 8 kHz
+    VSLCodecSpeex8000,
+    // Speex 16 kHz
+    VSLCodecSpeex16000,
+    // Speex 32 kHz
+    VSLCodecSpeex32000,
+    // GSM 8 kHZ
+    VSLCodecGSM,
+    // Opus
+    VSLCodecOpus
```
Comment (smolskyaleksey) [inline Pod/Classes/VSLCodecs.h]: there is no h264 codec

File: Pod/Classes/VSLCodecs.m [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
++ (BOOL)updateCodecs:(NSArray *)codecsToUse {
+    if ([VSLEndpoint sharedEndpoint].state != VSLEndpointStarted || [codecsToUse count] > 1) {
+        return NO;
+    }
+    
+    const unsigned codecInfoSize = 64;
+    pjsua_codec_info codecInfo[codecInfoSize];
+    unsigned codecCount = codecInfoSize;
+    pj_status_t status = pjsua_enum_codecs(codecInfo, &codecCount);
```
Comment (smolskyaleksey) [inline Pod/Classes/VSLCodecs.m]: H264 is video codec
```  obj-c
    const unsigned videoCodecInfoSize = 64;
    pjsua_codec_info videoCodecInfo[videoCodecInfoSize];
    unsigned videoCodecCount = videoCodecInfoSize;
    pj_status_t videoStatus = pjsua_vid_enum_codecs(videoCodecInfo, &videoCodecCount);
    if (videoStatus != PJ_SUCCESS) {
        DDLogError(@"Error getting list of video codecs");
        return NO;
    } else {
        for (NSUInteger i = 0; i < videoCodecCount; i++) {
            NSString *codecIdentifier = [NSString stringWithPJString:videoCodecInfo[i].codec_id];
            pj_uint8_t priority = [self priorityForVideoCodec:codecIdentifier];
            videoStatus = pjsua_vid_codec_set_priority(&videoCodecInfo[i].codec_id, priority);
            
            pjmedia_vid_codec_param param;
            pjsua_vid_codec_get_param(&videoCodecInfo[i].codec_id, &param);
            param.ignore_fmtp = PJ_TRUE;
            param.enc_fmt.det.vid.size.w = 288;
            param.enc_fmt.det.vid.size.h = 352;
            param.enc_fmt.det.vid.fps.num = 20;
            param.enc_fmt.det.vid.fps.denum = 1;
            param.dec_fmt.det.vid.size.w = 1920;
            param.dec_fmt.det.vid.size.h = 1920;
            pjsua_vid_codec_set_param(&videoCodecInfo[i].codec_id, &param);
            
            if (videoStatus != PJ_SUCCESS) {
                DDLogError(@"Error setting video codec priority to the correct value");
                return NO;
            }
        }
    }
  }
```
have to use methods for video codec.

File: Pod/Classes/VSLCodecs.h [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    // Speex 16 kHz
+    VSLCodecSpeex16000,
+    // Speex 32 kHz
+    VSLCodecSpeex32000,
+    // GSM 8 kHZ
+    VSLCodecGSM,
+    // Opus
+    VSLCodecOpus,
+    // H264 - Video codec
+    VSLCodecH264
```
Comment (smolskyaleksey) [inline Pod/Classes/VSLCodecs.h]: I think we should separate video and audio codecs to 2 enums

**Pull request closed without merging**